src/common/bip32.c

```diff
@@ -20,12 +20,20 @@ bool bip32_path_read(const uint8_t *in, size_t in_len, bip32_path_t *out) {
             return false;
         }
         out->path[i] = read_u32_be(in, offset);
-        if ((out->path[i] & 0x7FFFFFFFu) > MAX_DERIVATION_INDEX) {
+        offset += 4;
+    }
+    if (out->length > 4) {
```

Comment:
I feel we should add a description of the path. As far as I understood, it is "purpose / coin type / account / change / ....", right? Should we confirm that `purpose == 44'` and `coin_type == HATHOR_BIP44_CODE'`?

Should we force `account == 0`? Or should we allow up to 100`?

Reply:
The Makefile adds the `path` argument to the `APP_LOAD_PARAMS`, this argument is used at compile time to "lock" the base bip32 path this app is allowed. Whenever the OS derives the private key on a path it checks if it obeys this argument and throws an error if it doesn't, so we don't need to check for `purpose == 44'` or `coin_type == HATHOR_BIP44_CODE'`.

Reply:
Ok. Let's just add a comment explaining it, please.

Reply:
Great! Thanks!